### PR TITLE
Added doctrine factories

### PR DESCRIPTION
**Pull request by Nyholm**
It was doing a lot of copy-paste. Make sure I have not made any mistakes. 

Related issue: #8

### CODE REVIEW

File: src/Factory/DoctrineMemcachedFactory.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    /**
+     * {@inheritdoc}
+     */
+    protected static function configureOptionResolver(OptionsResolver $resolver)
+    {
+        $resolver->setDefaults([
+            'host' => '127.0.0.1',
+            'port' => '11211',
+        ]);
```
Comment (cryptiklemur) [inline src/Factory/DoctrineMemcachedFactory.php:43]: should this be on one line?

File: src/Factory/DoctrineMongodDBFactory.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    /**
+     * {@inheritdoc}
+     */
+    protected static function configureOptionResolver(OptionsResolver $resolver)
+    {
+        $resolver->setRequired([
+            'db',
+            'collection',
+        ]);
```
Comment (cryptiklemur) [inline src/Factory/DoctrineMongodDBFactory.php]: One line?

**Comment by cryptiklemur** [comment]
Needs a sync

File: src/Factory/DoctrineMongodDBFactory.php (hunk 2 above)
Comment (Nyholm) [inline src/Factory/DoctrineMongodDBFactory.php]: Sure, when doing setRequired we could use one one

**Comment by Nyholm** [comment]
Updated

**Pull request merged**